packages/blocks/src/api/registration.js

```diff
@@ -290,29 +238,15 @@ export function registerBlockType( blockNameOrMetadata, settings ) {
 		return;
 	}
 
-	if ( isObject( blockNameOrMetadata ) ) {
-		unstable__bootstrapServerSideBlockDefinitions( {
-			[ name ]: getBlockSettingsFromMetadata( blockNameOrMetadata ),
-		} );
-	}
+	const metadata = isObject( blockNameOrMetadata )
```

Comment:
I made this change because it's very useful for async block loading. At some places, like block inserter, my prototype looks at the server bootstrapped data, to avoid fully loading the block:
```js
const { title, parent, supports } = select( blocksStore ).getBootstrappedBlockType( 'core/query-pagination');
```

But this doesn't work for blocks that are registered only client-side, with no `block.json` metadata:
```js
registerBlockType( 'core/query-pagination', {
  title: __( 'Pagination' ),
  parent: [ 'core/query' ],
  supports: { inserter: false },
} );
```
There is no `metadata`, nothing is added to `state.bootstrappedBlockTypes` for this block. That's why I'm "reconstructing" the `metadata` by extracting selected fields from `settings`.

> the settings object can provide a different set of keys, and some of them can be already translated like `title` or `description`.

There are two scenarios that can happen here:
1. the `title` and `description` fields were bootstrapped from the server. Then they are localized (the server did that) and the call to `addBootstrappedBlock` in `registerBlockType` won't overwrite them. It will do at best the polyfilling of `.selectors`.
2. the fields were not bootstrapped from the server, and `addBoostrappedBlock` will add the data extracted from `settings` to `state.bootstrappedBlockType`. The `title` should be localized, because the JS code will call `title: __( 'Pagination' )` if the block is localized at all.

So, in neither of these cases are correct localized data overwritten with incorrent unlocalized data. We're just adding "fallback" bootstrapped metadata in case they don't exist at all.